coral-presto/src/main/java/com/linkedin/coral/presto/parser/ParseTreeBuilder.java

```diff
@@ -0,0 +1,1500 @@
+        return visitExternalBodyReference((ExternalBodyReference) node, context);
+      case "ExplainOption":
+        return visitExplainOption((ExplainOption) node, context);
+      case "ResetSession":
+        return visitResetSession((ResetSession) node, context);
+      case "SetSession":
+        return visitSetSession((SetSession) node, context);
+      case "ShowSession":
+        return visitShowSession((ShowSession) node, context);
+      case "Use":
+        return visitUse((Use) node, context);
+      case "ShowFunctions":
+        return visitShowFunctions((ShowFunctions) node, context);
+      case "ShowCreateFunction":
+        return visitShowCreateFunction((ShowCreateFunction) node, context);
+      case "ShowCreate":
+        return visitShowCreate((ShowCreate) node, context);
+      case "ShowStats":
+        return visitShowStats((ShowStats) node, context);
+      case "ShowColumns":
+        return visitShowColumns((ShowColumns) node, context);
+      case "ShowCatalogs":
+        return visitShowCatalogs((ShowCatalogs) node, context);
+      case "ShowSchemas ":
+        return visitShowSchemas((ShowSchemas) node, context);
+      case "ShowTables":
+        return visitShowTables((ShowTables) node, context);
+      case "Explain":
+        return visitExplain((Explain) node, context);
+      case "DescribeInput":
+        return visitDescribeInput((DescribeInput) node, context);
+      case "DescribeOutput":
+        return visitDescribeOutput((DescribeOutput) node, context);
+      case "Execute":
+        return visitExecute((Execute) node, context);
+      case "Deallocate":
+        return visitDeallocate((Deallocate) node, context);
+      case "Prepare":
+        return visitPrepare((Prepare) node, context);
+    }
+    throw getUnsupportedException(node);
+  }
+
+  private SqlNumericLiteral getNumericalLiteral(String literal) {
+    return SqlNumericLiteral.createExactNumeric(literal, ZERO);
+  }
+
+  /**
+   * It's used for most of the functions in the presto query. The unresolved call will be resolved in validation flow.
+   */
+  private SqlCall getUnresolvedFunction(String functionName, List<SqlNode> operands) {
+    SqlIdentifier functionIdentifier = createSqlIdentifier(SqlParserPos.ZERO, functionName);
+    return createCall(new SqlUnresolvedFunction(functionIdentifier, null, null, null, null, USER_DEFINED_FUNCTION),
+        operands);
+  }
+
+  private SqlCall getCall(SqlOperator operator, Node node, ParserVisitorContext context) {
+    if (operator == null) {
+      throw getUnsupportedException(node);
+    }
+    return operator.createCall(getParserPos(node), getChildSqlNodeList(node, context).getList());
+  }
+
+  @Override
+  protected SqlNode visitCurrentTime(CurrentTime node, ParserVisitorContext context) {
+    SqlParserPos pos = getParserPos(node);
+    switch (node.getFunction()) {
+      case TIME:
+        return createCall(CURRENT_TIME, Collections.emptyList(), pos);
```

Comment:
I plan to add function resolver that for the visitFunctionCall. For other visit function, since it's already specific, it doesn't have benefits to centralize them. In addition, if we create in each visit function we can keep more information such as position. Let me know what you think.